src/OVAL/oval_variable.c

```diff
@@ -781,9 +1013,39 @@ static xmlNode *_oval_VARIABLE_CONSTANT_to_dom(struct oval_variable *variable, x
 }
 
 static xmlNode *_oval_VARIABLE_EXTERNAL_to_dom(struct oval_variable *variable, xmlDoc * doc, xmlNode * parent) {
+
 	xmlNs *ns_definitions = xmlSearchNsByHref(doc, parent, OVAL_DEFINITIONS_NAMESPACE);
 	xmlNode *variable_node = xmlNewTextChild(parent, ns_definitions, BAD_CAST "external_variable", NULL);
 
+	struct oval_iterator *possible_values = oval_variable_get_possible_values(variable);
+	while (oval_collection_iterator_has_more(possible_values)) {
+		struct oval_variable_possible_value *pv = oval_collection_iterator_next(possible_values);
+		xmlNode *possible_value_node = xmlNewTextChild(variable_node, ns_definitions, BAD_CAST "possible_value", BAD_CAST pv->value);
+		xmlNewProp(possible_value_node, BAD_CAST "hint", BAD_CAST pv->hint);
+	}
+	oval_collection_iterator_free(possible_values);
+
+	struct oval_iterator *possible_restrictions = oval_variable_get_possible_restrictions(variable);
+	while (oval_collection_iterator_has_more(possible_restrictions)) {
+		struct oval_variable_possible_restriction *pr = oval_collection_iterator_next(possible_restrictions);
+		xmlNode *possible_restriction_node = xmlNewTextChild(variable_node, ns_definitions, BAD_CAST "possible_restriction", NULL);
```

Comment:
Yes, but such input would not pass the validation, because specification says that a <possible_restriction> element must have at least one <restriction> child element.